wgpu-hal/src/gles/queue.rs

```diff
@@ -559,8 +581,21 @@ impl super::Queue {
                     gl.disable_draw_buffer(glow::BLEND, draw_buffer);
                 }
             }
-            C::ClearColorF(draw_buffer, ref color) => {
-                gl.clear_buffer_f32_slice(glow::COLOR, draw_buffer, color);
+            C::ClearColorF {
+                draw_buffer,
+                ref color,
+                is_srgb,
+            } => {
+                if self
+                    .shared
+                    .private_caps
+                    .contains(super::PrivateCapabilities::WORKAROUND_MESA_I915_SRGB_SHADER_CLEAR)
+                    && is_srgb
+                {
+                    self.perform_shader_clear(draw_buffer, *color);
```

Comment:
I mentioned that we can't postpone `SetDrawBuffers`. My suggestion is the following:
The queue would have the "draw_buffer_count: u32" field as a part of the state, setup by `SetDrawBuffers` in addition to what it does today. Then `fn perform_shader_clear` can reset to this mask after it's done.